test/stdlib/TestJSONEncoder.swift

```diff
@@ -1615,6 +1671,7 @@ JSONEncoderTests.test("testEncodingTopLevelStructuredSingleClass") { TestJSONEnc
 JSONEncoderTests.test("testEncodingTopLevelDeepStructuredType") { TestJSONEncoder().testEncodingTopLevelDeepStructuredType()}
 JSONEncoderTests.test("testEncodingClassWhichSharesEncoderWithSuper") { TestJSONEncoder().testEncodingClassWhichSharesEncoderWithSuper() }
 JSONEncoderTests.test("testEncodingTopLevelNullableType") { TestJSONEncoder().testEncodingTopLevelNullableType() }
+JSONEncoderTests.test("testEncodingMultipleNestedContainersWithTheSameTopLevelKey") { TestJSONEncoder().testEncodingMultipleNestedContainersWithTheSameTopLevelKey() }
```

Comment:
We should add tests to ensure that re-encoding into different containers fails. Each test runs in its own process, so the expected fatal error will not take down the rest of the tests. I believe that the right way to assert that we expect a test to crash is `JSONEncoderTests.test("<test name here>").xfail(.always).code { TestJSONEncoder().<whatever> }`

Reply:
@jrose-apple Wasn't aware of that — is that new?
@pitiphong-p Sorry to have given bad info; we should probably do what Jordan suggests instead of XFAIL'ing the test.